Azure sso changes for both managed and unmanaged mariadb (#1)
ansible-middleware/azure-sso-demo: README.md

# azure-sso-demo

> This repository contains reference configurations that can create resourses on Azure using CLI and deploy cross-dc-rhsso-demo on Azure with all the required configurations.

Reviewer: Add a link to the [cross-dc-rhsso-demo](https://github.com/ansible-middleware/cross-dc-rhsso-demo) repository

Author: Added.

The final architecture looks like:

![Architecture diagram](./scenario.jpg)

> ## Prerequisites

Reviewer: Can we separate each line (especially those with code snippets) into a purpose and code snippet.

For example

* Install Ansible Dependencies
```
ansible-galaxy collection install -r requirements.yml
```

> * rhel 8 with registered subscription

Reviewer: ```suggestion
* Red Hat Enterprise Linux (RHEL) 8 host
```

> * ansible >= 2.9 / python >= 3.9

Reviewer: ```suggestion
* Ansible >= 2.9 / python >= 3.9
```

* `ansible-galaxy collection install -r requirements.yml`
* python3-netaddr installed on the controller host `dnf install python3-netaddr`
* Azure CLI installed on the controller host 'dnf install azure-cli'

> * All the default paramerts for Azure cloud instances are provided in 'roles/azure/defaults/main.yml'

Reviewer: ```suggestion
* All the default parmeters for Azure cloud instances are provided in `roles/azure/defaults/main.yml`
```

> * We are using dynamic inventory provided  by Azure. Once the instances are created you can view it by "ansible-inventory -i inventory/myazure_rm.yml --graph"

Reviewer: ```suggestion
* We are using dynamic inventory provided  by Azure. Once the instances are created you can view it by `ansible-inventory -i inventory/myazure_rm.yml --graph`
```

> * We have provided two options, we can either go with managed mariadb on Azure or we can create the RHEL instances and configuire mariadb using 'mariadb' role on cross-dc-rhsso-demo.

Reviewer: Break out into a "Database Options" bulletpoint

Author: Added:

### Database options

* We have provided two options: 
    1. For managed mariadb on Azure set managed_mariadb as true in `create-demo-setup.yml`.
    2. For unmanaged mariadb on Azure set managed_mariadb as false in `create-demo-setup.yml`. This will create the RHEL instances and configure mariadb using 'mariadb' role on cross-dc-rhsso-demo.

## Running on Azure cloud

1. Create a var-file containing your RHN credentials:
```
$ cat rhn-creds.yml
rhn_username: '<username>'
rhn_password: '<password>'
```

> 2. Execute the main play:

Reviewer: `provide_your_ssh_private_key` is not explained

Author: I think its self explained ? Shall I add a line as note ?

Author: Added:

Note: Replace `provide_your_ssh_private_key` with ssh private key file.

```
ansible-playbook -e @rhn-creds.yml -i cross-dc-rhsso-demo/myazure_rm.yml -i cross-dc-rhsso-demo/group_vars/all.yml -e "ansible_ssh_user=azureuser ansible_ssh_private_key_file='provide_your_ssh_private_key'" create-demo-setup.yml
```

## Security considerations

> * You can provide custom certificates for the role jbcs under (cross-dc-rhsso-demo/playbooks/roles/jbcs/files). If not provided it can create a self-signed certificates for SSO/JDG TLS connectivity.

Reviewer: Do we need to explain this in further detail?

Author: I taught it was good to have in readme. Its optional.

> 

Reviewer: What about a Validation section?

Author: Validation is added in the cross-dc repo.
